src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,109 @@
+import clsx from 'clsx';
+import React from 'react';
+import styles from './Badge.module.css';
+import Icon from '../Icon';
+import type { IconName, IconProps } from '../Icon';
+
+type BadgeProps = {
+  /**
+   * Text or icon to be placed on the upper right corner of the badgeable element.
+   */
+  children?: React.ReactNode;
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  // TODO-JL: appropriate prop name? i.e. variant?: 'empty' or empty?: boolean?
+  /**
+   * Empty variant of the badge with a smaller badge circle and no content.
+   */
+  empty?: boolean;
```

Comment:
As a result of converting the common `<Badge>` subcomponent to `<Badge.Icon>` and `<Badge.Text>`, the need for the `empty` variant case rises to support the smaller badge without any content. This can be solved with, and is also supported with `<Badge.Text />` but using the `Text` subcomponent seems unintuitive.

Please confirm whether this variant is necessary, and if so, should use `empty?: boolean` or `variant?: 'empty'`

Reply:
I feel like that's a little odd because the other variants pass in a subcomponent but this one would use a prop. Would something like `<Badge.Empty /> or `<Badge.Dot />` be too weird?

Reply:
I'm with @dierat and think something like `.Dot` could actually work nicely. Internally, it could be it's own sub-component, or just `Badge.Text` with a non-breaking or no space in its content (or some other thing). for these "types", they are just a convenience API we offer :)

Then we could do away with the prop.

People can then do something like:

```
{ count > 0 ? (
    <Badge.Text>{count}</Badge.Text>
) : (
    <Badge.Empty />
)}
```

if needed.

Reply:
Created `<Badge.Dot>` as suggested and removed `empty` prop